zgw/object-management/src/main/kotlin/com/ritense/objectmanagement/service/ObjectManagementService.kt

```diff
@@ -121,35 +129,15 @@ class ObjectManagementService(
         pageable: Pageable
     ): PageImpl<ObjectsListRowDto> {
         val objectManagement = getById(id) ?: let {
-            logger.info {
-                "The requested Id is not configured as a objectnamagement configuration. " +
-                    "The requested id was: $id"
-            }
-            throw NotFoundException()
+            throw IllegalStateException("The requested Id is not configured as a object management configuration. The requested id was: $id")
         }
 
         val searchFieldList = searchFieldV2Service.findAllByOwnerId(id.toString())!!
 
         val searchDtoList = searchFieldList.flatMap { searchField ->
             searchWithConfigRequest.otherFilters
                 .filter { otherFilter -> otherFilter.key == searchField.key }
-                .flatMap { otherFilter ->
-                    if (searchField.fieldType != FieldType.RANGE) {
-                        if (otherFilter.values.size > 1) {
-                            throw IllegalArgumentException("The objects api does not support the multiselect options")
-                        }
-                        otherFilter.values.flatMap { value ->
-                            mapToObjectSearchParameter(searchField.key, searchField.fieldType, value.value, searchField.dataType)
-                        }
-                    } else {
-                        mapToObjectSearchParameter(
-                            searchField.key,
-                            searchField.fieldType,
-                            listOf(otherFilter.rangeFrom?.value, otherFilter.rangeTo?.value),
-                            searchField.dataType
-                        )
-                    }
-                }
+                .flatMap { otherFilter -> mapToObjectSearchParameters(searchField, otherFilter) }
         }
 
         val searchString = ObjectSearchParameter.toQueryParameter(searchDtoList)
```

Comment:
I just noticed we don't do sanitation when converting values to query parameters. I think we should url-encode them.

Reply:
I just looked it up: a minimal amount of url encoding is done by default in the webclient when adding this searchstring as a query parameter. [docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/UriBuilder.html#queryParam(java.lang.String,java.lang.Object...))